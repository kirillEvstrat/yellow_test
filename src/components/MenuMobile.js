import React from "react";
import {NavLink} from "react-router-dom";
import Menu from "../images/menu.png";
import close from "../images/close.png";
import logo from "../images/mobileMenulofo/logo.png";

class MenuMobile extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            isOpen: false
        }
    };

    openHandler = (e) => {
        this.setState( state => ({...state, isOpen: !state.isOpen}));
    };

    render() {
        return (
            !this.state.isOpen ?
                (
                    <img className='menu-open' src={Menu} alt="" onClick={e => this.openHandler(e)}/>
                ) :
                (
                <div className='mobile-menu-wr'>
                    <img className='menu-logo' src={logo} alt=""/>
                    <img className='menu-close' src={close} alt="" onClick={e => this.openHandler(e)}/>
                    <div className='mobile-menu-items'>
                        <div className='mobile-menu-item'>
                            <NavLink to='/jogs'>JOGS</NavLink>
                        </div>
                        <div className='mobile-menu-item'>
                            <NavLink to='/info'>INFO</NavLink>
                        </div>
                        <div className='mobile-menu-item'>
                            <NavLink to='/contact'>CONTACT US</NavLink>
                        </div>
                    </div>
                </div>
                )
        )
    };
}

export default MenuMobile;