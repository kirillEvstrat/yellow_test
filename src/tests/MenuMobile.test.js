import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";
import MenuMobile from "../components/MenuMobile";

test('close/open mobile menu with routing', () => {
    const component = renderer.create(
        <BrowserRouter>
            <MenuMobile />
        </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});