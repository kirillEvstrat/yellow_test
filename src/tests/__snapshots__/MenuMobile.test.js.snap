// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`close/open mobile menu with routing 1`] = `
<img
  alt=""
  className="menu-open"
  onClick={[Function]}
  src="menu.png"
/>
`;

exports[`close/open mobile menu with routing 2`] = `
<div
  className="mobile-menu-wr"
>
  <img
    alt=""
    className="menu-logo"
    src="logo.png"
  />
  <img
    alt=""
    className="menu-close"
    onClick={[Function]}
    src="close.png"
  />
  <div
    className="mobile-menu-items"
  >
    <div
      className="mobile-menu-item"
    >
      <a
        aria-current={null}
        href="/jogs"
        onClick={[Function]}
      >
        JOGS
      </a>
    </div>
    <div
      className="mobile-menu-item"
    >
      <a
        aria-current={null}
        href="/info"
        onClick={[Function]}
      >
        INFO
      </a>
    </div>
    <div
      className="mobile-menu-item"
    >
      <a
        aria-current={null}
        href="/contact"
        onClick={[Function]}
      >
        CONTACT US
      </a>
    </div>
  </div>
</div>
`;

exports[`close/open mobile menu with routing 3`] = `
<div
  className="mobile-menu-wr"
>
  <img
    alt=""
    className="menu-logo"
    src="logo.png"
  />
  <img
    alt=""
    className="menu-close"
    onClick={[Function]}
    src="close.png"
  />
  <div
    className="mobile-menu-items"
  >
    <div
      className="mobile-menu-item"
    >
      <a
        aria-current={null}
        href="/jogs"
        onClick={[Function]}
      >
        JOGS
      </a>
    </div>
    <div
      className="mobile-menu-item"
    >
      <a
        aria-current={null}
        href="/info"
        onClick={[Function]}
      >
        INFO
      </a>
    </div>
    <div
      className="mobile-menu-item"
    >
      <a
        aria-current={null}
        href="/contact"
        onClick={[Function]}
      >
        CONTACT US
      </a>
    </div>
  </div>
</div>
`;
